### ISSUE TITLE
could you please fix the audio for ALC668..layout 1 wont work anymore. 

### ISSUE THREAD

**Issue by neek619** [syscl/M3800#98]

**Comment by starfireprime** [comment]
LOL - that's the only thing I can see not working after the Mojave update.  

Syscl - first thanks for your work!  Can you point me at some background info related to the audio driver?  I see that you have CodecCommander in CLOVER and five patches for ALC668.  I'd like to know more about how you configured the system in the past.  Thanks!

Edit:
Ok - I was able to get audio working in Mojave by 1) rolling back some of the audio changes in syscl's scripts and 2) using the newest AppleALC and Lilu kexts.

Prerequisites:
-  Updated M3800 git pull
-  Updated Clover Configurator (5.2.0.1) 
-  Newest AppleALC.kext -> version 1.3.2
- Newest Lilu.kext -> version 1.2.7
-  Presumed that you have run Deploy.sh at least once after upgrading to Mojave, but sound does not work.

Procedure (what I did):

1) Update kext in EFI/CLOVER/kexts/Other/Lilu.kext.  Reboot to make sure all is well.

2) Remove syscl audio mods
  a) Copy Deploy.sh to Deploy_noaudio.sh and chmod 775 Deploy_noaudio.sh
  b) Comment out lines that
      1) Copies AppleHDA_ALC668.kext to /Library/Extensions (Line 412)
      2) Patches the DSDT with audio_HDEF-layout1 (Line 1509)

3) Before executing the modified script, delete AppleHDA_ALC668.kext from /L/E

4) Execute the modified script.  Check that AppleHDA_ALC668.kext is not in /L/E.  Reboot to make sure all is well.

5) Now we are going to apply the "standard" AppleALC mods in CLOVER.

  a) Mount EFI partition 

  b) In EFI/CLOVER/kexts/Other
     1) Remove CodecCommander.kext
     2) Copy in new AppleALC.kext

  c) Start CloverConfigurator and open your EFI/CLOVER/config.plist
    1) Panel Acpi / DSDT tab : Under "List Of Patches" add "change HDAS to HDEF"
    2) Panel Devices : Upper right, check "ResetHDA" and in the Inject field to the left enter 20
    3) Panel Kernel and Kext Patches : Disable all AppleHDA entries (I left the AppleHDAController patches)
    4) Save the new config.plist

6) Reboot and verify that the sound device is visible and sound effects are audible.

**Comment by syscl** [comment]
I’m currently working on it and it did work on M3800. Wait for the PR to be merged.

**Comment by starfireprime** [comment]
Thanks!  Please see my edited post above.

**Comment by syscl** [comment]
As I said, I've PR to AppleALC upstream already and no more modification will be needed once the PR get accepted.

**Comment by starfireprime** [comment]
Great!  I'll keep an eye out for the update.  Thanks again!

**Comment by neek619** [comment]
will wait for the update..
for now I am using voodoohda.kext for enabling sound.

**Comment by GuillaumeHullin** [comment]
@starfireprime do you have also your headphone output working? I just did your procedure and I successfully get the speaker to work but when I plug headphone there is some kind of pink noise for half a sec and nothing after.
I used Lilu 1.2.8 and AppleALC 1.3.3

**Comment by starfireprime** [comment]
@GuillaumeHullin 
Sorry - I never tried the headphones as I never use them...

I see in the the AppleALC repository that syscl uploaded a fix to Layout 27 for our laptops.  I switched to 27 in CLOVER and the headphone jack is recognized and works, but only for a minute or so.  After that there's an awful white noise build up and then the headphones do not work.  (This was the same for layout 29 for me as well).

I am using AppleALC1.3.3 as well.  Although the changelog does not mention the update, looking into the Info.plist in the 1.3.3  source tarball, syscl's update does appear to be in that release.

Edit:
One more piece of info - after the static crescendo the headphone sound stops working.  But if I switch the "Input" from internal microphone to line in or vice verse, then headphone sound returns for another minute or so until the static returns.  I  don't know the intricacies of sharing an input/output jack, but I'm guessing that a background process is periodically running that creates a conflict on the headphone output.

**Comment by GuillaumeHullin** [comment]
@starfireprime I see... well I guess I'm gonna wait for an update later. What kind of layer are you talking about (27, 29).... I don't understand this, maybe you could point me to some source that would explain a bit more ;) (Google wasnt too friendly with me this time ahaha)

**Comment by starfireprime** [comment]
@GuillaumeHullin 
I was using the ALC668 entry in the table here [https://github.com/acidanthera/AppleALC/wiki/Supported-codecs](url) to try different layout ids.  When I saw syscl's modification, I presume that 27 is the one we want to use.  I do not know what the 0x100003 entry in the table is referring to.  Perhaps that's a missing piece.  Please see my edit above concerning the selection of input line that temporarily restores headphone output.

**Comment by GuillaumeHullin** [comment]
Ok so... I was playing around and honestly I don't know what I exactly did.... 
In Clover configurator I changed the inject value from 20 to 29... reboot and it worked... the headphones worked without stopping for several minutes. However I could hear some kind of cracking in the audio... in the headphones and in the speakers. But the headphones were working.
So I tried an other number 30, 26, 27, 28.... no better result... so I put back 29. And then... headphones doesn't work anymore...

**Comment by starfireprime** [comment]
I played with it a little more and think I have it sorted out.  Using the current AppleALC (1.3.3) and Lilu (1.2.8) and injected 27 into the Audio on the Devices tab of CloverConfigurator.  I don't have ResetHDA checked - don't know if it matters.

I think my problem was that I did not disable the AppleHDAController kext patches.  I had disabled the plain AppleHDA patches, but once I had all of them disabled, then I could use the headphones and I have yet to hear the static crescendo.

**Comment by GuillaumeHullin** [comment]
Yup!! It's working! In my case ResetHDA is checked... mustnt matter much :)
However I still hear some crack in the sound... nothing too much but as a video editor I can spot it. I'll boot on Windows to check if there is a difference.

**Comment by GuillaumeHullin** [comment]
@starfireprime Goshhh.... I still got problem with the headphones... I was doing some modification in my ktext and DSDTs and then I lost my audio on the headphone.... I restored my EFI to the backup I made when headphones were working... nope!
I'm trying to find out what happened but I can't manage to figure out. I'm thinking there is something that changed on my OS itself... but no.
Any idea?
